functests/1_performance/cpu_management.go

```diff
@@ -483,9 +511,78 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			}
 		})
 	})
+
+	When("strict NUMA aligment is requested", func() {
+		var testpod *corev1.Pod
+
+		BeforeEach(func() {
+			if profile.Spec.NUMA == nil || profile.Spec.NUMA.TopologyPolicy == nil {
+				Skip("Topology Manager Policy is not configured")
+			}
+			tmPolicy := *profile.Spec.NUMA.TopologyPolicy
+			if tmPolicy != "single-numa-node" {
+				Skip("Topology Manager Policy is not Single NUMA Node")
+			}
+		})
+
+		AfterEach(func() {
+			if testpod == nil {
+				return
+			}
+			deleteTestPod(testpod)
+		})
+
+		It("should reject pods which request integral CPUs not aligned with machine SMT level", func() {
+			cpuCount := 1 // not must be intentionally < than the smtLevel
+			testpod = promotePodToGuaranteed(getStressPod(workerRTNode.Name, cpuCount))
+			testpod.Namespace = testutils.NamespaceTesting
+
+			smtLevel := getSMTLevel(workerRTNode)
+			if smtLevel < 2 {
```

Comment:
does it make to run skip before pod definition?

Reply:
yes and we can probably mode this check right at the beginning